core/include/traccc/fitting/kf_algorithm.hpp

```diff
@@ -0,0 +1,76 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022-2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/edm/track_candidate.hpp"
+#include "traccc/edm/track_state.hpp"
+#include "traccc/fitting/fitting_config.hpp"
+#include "traccc/geometry/detector.hpp"
+#include "traccc/utils/algorithm.hpp"
+
+// Detray include(s).
+#include <detray/detectors/bfield.hpp>
+
+namespace traccc::host {
+
+/// Kalman filter based track fitting algorithm
+class kf_algorithm : public algorithm<track_state_container_types::host(
```

Comment:
This should be just called `kalman_fitter` instead of `kf_algorithm`

Reply:
The `kalman_fitter` fitter class still exists.

🤔 We could call it `traccc::host::kalman_fitter_algorithm` instead, but since `traccc::host::ckf_algorithm` already went through, I thought I'd go with this short name here as well.

Reply:
This is what is exposed to the user directly and naming it into `kf_algorithm` feels less obvious to me. `kalman_fitting_algorithm` might work better
`ckf_algorithm` probably need to be changed to `ckf_finding_algorithm`  or sth else